src/backends/cmake.rs

```diff
@@ -1,68 +1,98 @@
-    }
+use crate::backends::{
+    BatchBackend, BatchBuildResults, BuildCommandOptions, BuildProfile, BuildResult, CommandSpec,
+};
 
-    fn build(&self, config: &BuildArgs) -> io::Result<()> {
-        fs::create_dir_all(format!("{}/build", self.lfc.out.display()))?;
+pub struct Cmake;
 
-        // cmake generation
-        let mut cmake_command = Command::new("cmake");
-        cmake_command.arg(format!(
-            "-DCMAKE_BUILD_TYPE={}",
-            if config.release { "RELEASE" } else { "DEBUG" }
-        ));
-        cmake_command.arg(format!("-DCMAKE_INSTALL_PREFIX={}", self.lfc.out.display()));
-        cmake_command.arg("-DCMAKE_INSTALL_BINDIR=bin");
-        cmake_command.arg("-DREACTOR_CPP_VALIDATE=ON");
-        cmake_command.arg("-DREACTOR_CPP_TRACE=OFF");
-        cmake_command.arg("-DREACTOR_CPP_LOG_LEVEL=3");
-        cmake_command.arg(format!(
-            "-DLF_SRC_PKG_PATH={}",
-            self.app.root_path.display()
-        ));
-        cmake_command.arg(format!("{}/src-gen", self.lfc.out.display()));
-        cmake_command.arg(format!("-B {}/build", self.lfc.out.display()));
-        cmake_command.current_dir(format!("{}/build", self.lfc.out.display()));
-        run_and_capture(&mut cmake_command)?;
+fn gen_cmake_files(app: &App, options: &BuildCommandOptions) -> BuildResult {
+    let build_dir = app.output_root.join("build");
+    fs::create_dir_all(&build_dir)?;
 
-        // compiling
-        let mut cmake_build_command = Command::new("cmake");
-        cmake_build_command.current_dir(format!("{}/build", self.lfc.out.display()));
-        cmake_build_command.arg("--build");
-        cmake_build_command.arg("./");
-        run_and_capture(&mut cmake_build_command)?;
+    let mut cmake = Command::new("cmake");
+    cmake.arg(format!(
+        "-DCMAKE_BUILD_TYPE={}",
+        if options.profile == BuildProfile::Release {
+            "RELEASE"
+        } else {
+            "DEBUG"
+        }
+    ));
+    cmake.arg(format!(
+        "-DCMAKE_INSTALL_PREFIX={}",
+        app.output_root.display()
+    ));
+    cmake.arg("-DCMAKE_INSTALL_BINDIR=bin");
+    cmake.arg("-DREACTOR_CPP_VALIDATE=ON");
+    cmake.arg("-DREACTOR_CPP_TRACE=OFF");
+    cmake.arg("-DREACTOR_CPP_LOG_LEVEL=3");
+    cmake.arg(format!("-DLF_SRC_PKG_PATH={}", app.root_path.display()));
+    cmake.arg(app.src_gen_dir());
+    cmake.arg(format!("-B {}", build_dir.display()));
+    cmake.current_dir(&build_dir);
 
-        // installing
-        let mut cmake_install_command = Command::new("cmake");
-        cmake_install_command.current_dir(format!("{}/build", self.lfc.out.display()));
-        cmake_install_command.arg("--install");
-        cmake_install_command.arg("./");
-        run_and_capture(&mut cmake_install_command)?;
+    execute_command_to_build_result(cmake)
+}
 
-        Ok(())
+fn do_cmake_build(results: &mut BatchBuildResults, options: &BuildCommandOptions) {
+    super::lfc::LFC::do_parallel_lfc_codegen(options, results, false);
```

Comment:
why do we invoke the code generation here ? Then every backend would need to invoke the source generator. 

And its not nicely separated into layers (code gen -> tool chain)